addon/validators/number.js

```diff
@@ -49,16 +49,12 @@ function _validateType(type, opts, numValue, key) {
 
 export default function validateNumber(options = {}) {
   return (key, value) => {
-    let numValue = Number(value);
-    let optKeys = keys(options);
-
-    if (options.allowBlank && isEmpty(value)) {
-      return true;
+    if (isEmpty(value)) {
```

Comment:
Could we do `typeOf(value) !== 'number'` instead? That feels a little more explicit and the validation message is more likely to make sense. Although we might also need to check if it's `NaN` since that is also of type number